net/dnsfallback/dnsfallback.go

```diff
@@ -76,11 +76,27 @@ func MakeLookupFunc(logf logger.Logf, netMon *netmon.Monitor) func(ctx context.C
 				metricRecursiveErrors.Add(1)
 				return
 			}
-			slices.SortFunc(addrs, func(a, b netip.Addr) bool { return a.Less(b) })
+			slices.SortFunc(addrs, func(a, b netip.Addr) int {
+				if a.Less(b) {
+					return -1
+				} else if b.Less(a) {
+					return 1
+				} else {
+					return 0
+				}
+			})
 
 			// Wait for a response from the main function
 			oldAddrs := <-addrsCh
-			slices.SortFunc(oldAddrs, func(a, b netip.Addr) bool { return a.Less(b) })
+			slices.SortFunc(oldAddrs, func(a, b netip.Addr) int {
+				if a.Less(b) {
+					return -1
+				} else if b.Less(a) {
+					return 1
+				} else {
+					return 0
```

Comment:
amusingly we used to have Compare and as part of putting it in std, they asked us to remove it: https://pkg.go.dev/inet.af/netaddr#IP.Compare

Now we have evidence we need it back :)

Reply:
Should we add AddrCompare to https://pkg.go.dev/go4.org/netipx ?

Reply:
https://github.com/go4org/netipx/pull/14